File: topics/procedure-decl/exercises/SceneDrawing.pas
program GameMain;
uses sgGraphics, sgUtils, sgResources;

procedure Main();
begin
    OpenGraphicsWindow('Long Code', 800, 600);
    
    //First Background
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();
    
    Delay(1000);
    
    //Second Sun
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();

    Delay(200);

    //Draw the sun
    FillCircle(ColorYellow, 50.0, 50.0, 20);
    DrawCircle(ColorRed, 50.0, 50.0, 15);
    DrawCircle(ColorRed, 50.0, 50.0, 12);
    DrawCircle(ColorRed, 50.0, 50.0, 9);
    DrawCircle(ColorRed, 50.0, 50.0, 6);
    FillCircle(ColorWhite, 50.0, 50.0, 3);
    DrawLine(ColorYellow, 50, 75, 50, 110);
    DrawLine(ColorYellow, 25, 75, 0, 100);
    DrawLine(ColorYellow, 75, 75, 100, 100);
    DrawLine(ColorYellow, 0, 50, 25, 50);
    DrawLine(ColorYellow, 75, 50, 100, 50);
    DrawLine(ColorYellow, 25, 25, 0, 0);
    DrawLine(ColorYellow, 75, 25, 100, 0);
    DrawLine(ColorYellow, 50, 0, 50, 25);
    RefreshScreen();

    Delay(1000);

    //Third Face
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();
    Delay(200);
    
    //Draw the sun
    FillCircle(ColorYellow, 50.0, 50.0, 20);
    DrawCircle(ColorRed, 50.0, 50.0, 15);
    DrawCircle(ColorRed, 50.0, 50.0, 12);
    DrawCircle(ColorRed, 50.0, 50.0, 9);
    DrawCircle(ColorRed, 50.0, 50.0, 6);
    FillCircle(ColorWhite, 50.0, 50.0, 3);
    DrawLine(ColorYellow, 50, 75, 50, 110);
    DrawLine(ColorYellow, 25, 75, 0, 100);
    DrawLine(ColorYellow, 75, 75, 100, 100);
    DrawLine(ColorYellow, 0, 50, 25, 50);
    DrawLine(ColorYellow, 75, 50, 100, 50);
    DrawLine(ColorYellow, 25, 25, 0, 0);
    DrawLine(ColorYellow, 75, 25, 100, 0);
    DrawLine(ColorYellow, 50, 0, 50, 25);
    RefreshScreen();
    Delay(200);

    //Draw Face
    FillCircle(ColorPink, 100, 250, 80);
    FillEllipse(ColorWhite, 40, 240, 40, 30);
    FillEllipse(ColorWhite, 120, 240, 40, 30);
    FillCircle(ColorBlue, 67, 260, 10);
    FillCircle(ColorBlue, 132, 260, 10);        
    FillTriangle(ColorRed, 100, 240, 85, 280, 115, 280);    
    DrawTriangle(ColorBlack, 100, 240, 85, 280, 115, 280);  
    FillTriangle(ColorRed, 100, 290, 100, 310, 70, 300);    
    FillTriangle(ColorRed, 100, 290, 100, 310, 130, 300);   
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 70, 300);  
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 130, 300); 

    RefreshScreen();

    Delay(1000);
    
    //Fourth House
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();
    Delay(200);
    
    //Draw the sun
    FillCircle(ColorYellow, 50.0, 50.0, 20);
    DrawCircle(ColorRed, 50.0, 50.0, 15);
    DrawCircle(ColorRed, 50.0, 50.0, 12);
    DrawCircle(ColorRed, 50.0, 50.0, 9);
    DrawCircle(ColorRed, 50.0, 50.0, 6);
    FillCircle(ColorWhite, 50.0, 50.0, 3);
    DrawLine(ColorYellow, 50, 75, 50, 110);
    DrawLine(ColorYellow, 25, 75, 0, 100);
    DrawLine(ColorYellow, 75, 75, 100, 100);
    DrawLine(ColorYellow, 0, 50, 25, 50);
    DrawLine(ColorYellow, 75, 50, 100, 50);
    DrawLine(ColorYellow, 25, 25, 0, 0);
    DrawLine(ColorYellow, 75, 25, 100, 0);
    DrawLine(ColorYellow, 50, 0, 50, 25);
    RefreshScreen();
    Delay(200);

    //Draw Face
    FillCircle(ColorPink, 100, 250, 80);
    FillEllipse(ColorWhite, 40, 240, 40, 30);
    FillEllipse(ColorWhite, 120, 240, 40, 30);
    FillCircle(ColorBlue, 67, 260, 10);
    FillCircle(ColorBlue, 132, 260, 10);        
    FillTriangle(ColorRed, 100, 240, 85, 280, 115, 280);    
    DrawTriangle(ColorBlack, 100, 240, 85, 280, 115, 280);  
    FillTriangle(ColorRed, 100, 290, 100, 310, 70, 300);    
    FillTriangle(ColorRed, 100, 290, 100, 310, 130, 300);   
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 70, 300);  
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 130, 300); 
    RefreshScreen();
    Delay(200);

    //Draw House
    FillTriangle(ColorRed, 300, 160, 200, 260, 400, 260);
    FillRectangle(ColorGrey, 210, 260, 180, 180);
    FillRectangle(ColorBlack, 215, 300, 80, 140);
    FillRectangle(ColorLightGrey, 300, 300, 80, 60);
    FillTriangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    FillTriangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    FillCircle(ColorYellow, 280, 370, 3);
    RefreshScreen();

    Delay(1000);
    
    //Fifth Tree
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();
    Delay(200);
    
    //Draw the sun
    FillCircle(ColorYellow, 50.0, 50.0, 20);
    DrawCircle(ColorRed, 50.0, 50.0, 15);
    DrawCircle(ColorRed, 50.0, 50.0, 12);
    DrawCircle(ColorRed, 50.0, 50.0, 9);
    DrawCircle(ColorRed, 50.0, 50.0, 6);
    FillCircle(ColorWhite, 50.0, 50.0, 3);
    DrawLine(ColorYellow, 50, 75, 50, 110);
    DrawLine(ColorYellow, 25, 75, 0, 100);
    DrawLine(ColorYellow, 75, 75, 100, 100);
    DrawLine(ColorYellow, 0, 50, 25, 50);
    DrawLine(ColorYellow, 75, 50, 100, 50);
    DrawLine(ColorYellow, 25, 25, 0, 0);
    DrawLine(ColorYellow, 75, 25, 100, 0);
    DrawLine(ColorYellow, 50, 0, 50, 25);
    RefreshScreen();
    Delay(200);

    //Draw Face
    FillCircle(ColorPink, 100, 250, 80);
    FillEllipse(ColorWhite, 40, 240, 40, 30);
    FillEllipse(ColorWhite, 120, 240, 40, 30);
    FillCircle(ColorBlue, 67, 260, 10);
    FillCircle(ColorBlue, 132, 260, 10);        
    FillTriangle(ColorRed, 100, 240, 85, 280, 115, 280);    
    DrawTriangle(ColorBlack, 100, 240, 85, 280, 115, 280);  
    FillTriangle(ColorRed, 100, 290, 100, 310, 70, 300);    
    FillTriangle(ColorRed, 100, 290, 100, 310, 130, 300);   
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 70, 300);  
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 130, 300); 
    RefreshScreen();
    Delay(200);

    //Draw House
    FillTriangle(ColorRed, 300, 160, 200, 260, 400, 260);
    FillRectangle(ColorGrey, 210, 260, 180, 180);
    FillRectangle(ColorBlack, 215, 300, 80, 140);
    FillRectangle(ColorLightGrey, 300, 300, 80, 60);
    FillTriangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    FillTriangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    FillCircle(ColorYellow, 280, 370, 3);
    RefreshScreen();
    Delay(200);

    //Draw Tree
    FillTriangle(ColorGreen, 500, 75, 440, 150, 560, 150);          
    FillEllipse(ColorTurquoise, 400, 380, 200, 100);
    FillRectangle(ColorBlack, 495, 274, 20, 160);
    FillTriangle(ColorGreen, 500, 120, 400, 275, 600, 275);
    FillCircle(ColorRed, 490, 150.0, 3);
    FillCircle(ColorRed, 510, 170, 4);
    FillCircle(ColorRed, 480, 240, 3);
    FillCircle(ColorRed, 560, 260, 5);
    RefreshScreen();

    Delay(1000);
    
    //Sixth Flats
    //Draw the background
    FillRectangle(ColorBlue, 0, 0, 800, 300);
    FillRectangle(ColorGreen, 0, 300, 800, 300);
    RefreshScreen();
    Delay(200);
    
    //Draw the sun
    FillCircle(ColorYellow, 50.0, 50.0, 20);
    DrawCircle(ColorRed, 50.0, 50.0, 15);
    DrawCircle(ColorRed, 50.0, 50.0, 12);
    DrawCircle(ColorRed, 50.0, 50.0, 9);
    DrawCircle(ColorRed, 50.0, 50.0, 6);
    FillCircle(ColorWhite, 50.0, 50.0, 3);
    DrawLine(ColorYellow, 50, 75, 50, 110);
    DrawLine(ColorYellow, 25, 75, 0, 100);
    DrawLine(ColorYellow, 75, 75, 100, 100);
    DrawLine(ColorYellow, 0, 50, 25, 50);
    DrawLine(ColorYellow, 75, 50, 100, 50);
    DrawLine(ColorYellow, 25, 25, 0, 0);
    DrawLine(ColorYellow, 75, 25, 100, 0);
    DrawLine(ColorYellow, 50, 0, 50, 25);
    RefreshScreen();
    Delay(200);

    //Draw Face
    FillCircle(ColorPink, 100, 250, 80);
    FillEllipse(ColorWhite, 40, 240, 40, 30);
    FillEllipse(ColorWhite, 120, 240, 40, 30);
    FillCircle(ColorBlue, 67, 260, 10);
    FillCircle(ColorBlue, 132, 260, 10);        
    FillTriangle(ColorRed, 100, 240, 85, 280, 115, 280);    
    DrawTriangle(ColorBlack, 100, 240, 85, 280, 115, 280);  
    FillTriangle(ColorRed, 100, 290, 100, 310, 70, 300);    
    FillTriangle(ColorRed, 100, 290, 100, 310, 130, 300);   
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 70, 300);  
    DrawTriangle(ColorBlack, 100, 290, 100, 310, 130, 300); 
    RefreshScreen();
    Delay(200);

    //Draw House
    FillTriangle(ColorRed, 300, 160, 200, 260, 400, 260);
    FillRectangle(ColorGrey, 210, 260, 180, 180);
    FillRectangle(ColorBlack, 215, 300, 80, 140);
    FillRectangle(ColorLightGrey, 300, 300, 80, 60);
    FillTriangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    FillTriangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    FillCircle(ColorYellow, 280, 370, 3);
    RefreshScreen();
    Delay(200);

    //Draw Tree
    FillTriangle(ColorGreen, 500, 75, 440, 150, 560, 150);          
    FillEllipse(ColorTurquoise, 400, 380, 200, 100);
    FillRectangle(ColorBlack, 495, 274, 20, 160);
    FillTriangle(ColorGreen, 500, 120, 400, 275, 600, 275);
    FillCircle(ColorRed, 490, 150.0, 3);
    FillCircle(ColorRed, 510, 170, 4);
    FillCircle(ColorRed, 480, 240, 3);
    FillCircle(ColorRed, 560, 260, 5);
    RefreshScreen();
    Delay(200);

    //Draw the block
    FillRectangle(ColorRed, 620, 50, 160, 450);
    FillRectangle(ColorBlack, 640, 75, 40, 60);
    FillRectangle(ColorLightGrey, 720, 75, 40, 60);
    FillRectangle(ColorGrey, 640, 225, 40, 60);
    FillRectangle(ColorYellow, 720, 225, 40, 60);
    FillRectangle(ColorBlack, 670, 380, 60, 120);
    RefreshScreen();
    
    Delay(2000);
    
    ReleaseAllResources();
end;

begin
    Main();
end.

File: topics/procedure-decl/exercises/scene-drawing.c
#include "SwinGame.h"

int main(int argc, char* argv[])
{
    open_graphics_window("Long Code", 800, 600);
    load_default_colors();
    
    //First Background
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();
    
    delay(1000);
    
    //Second Sun
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();

    delay(200);

    //Draw the sun
    fill_circle(ColorYellow, 50.0, 50.0, 20);
    draw_circle(ColorRed, 50.0, 50.0, 15);
    draw_circle(ColorRed, 50.0, 50.0, 12);
    draw_circle(ColorRed, 50.0, 50.0, 9);
    draw_circle(ColorRed, 50.0, 50.0, 6);
    fill_circle(ColorWhite, 50.0, 50.0, 3);
    draw_line(ColorYellow, 50, 75, 50, 110);
    draw_line(ColorYellow, 25, 75, 0, 100);
    draw_line(ColorYellow, 75, 75, 100, 100);
    draw_line(ColorYellow, 0, 50, 25, 50);
    draw_line(ColorYellow, 75, 50, 100, 50);
    draw_line(ColorYellow, 25, 25, 0, 0);
    draw_line(ColorYellow, 75, 25, 100, 0);
    draw_line(ColorYellow, 50, 0, 50, 25);
    refresh_screen();

    delay(1000);

    //Third Face
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();
    delay(200);
    
    //Draw the sun
    fill_circle(ColorYellow, 50.0, 50.0, 20);
    draw_circle(ColorRed, 50.0, 50.0, 15);
    draw_circle(ColorRed, 50.0, 50.0, 12);
    draw_circle(ColorRed, 50.0, 50.0, 9);
    draw_circle(ColorRed, 50.0, 50.0, 6);
    fill_circle(ColorWhite, 50.0, 50.0, 3);
    draw_line(ColorYellow, 50, 75, 50, 110);
    draw_line(ColorYellow, 25, 75, 0, 100);
    draw_line(ColorYellow, 75, 75, 100, 100);
    draw_line(ColorYellow, 0, 50, 25, 50);
    draw_line(ColorYellow, 75, 50, 100, 50);
    draw_line(ColorYellow, 25, 25, 0, 0);
    draw_line(ColorYellow, 75, 25, 100, 0);
    draw_line(ColorYellow, 50, 0, 50, 25);
    refresh_screen();
    delay(200);

    //Draw Face
    fill_circle(ColorPink, 100, 250, 80);
    fill_ellipse(ColorWhite, 40, 240, 40, 30);
    fill_ellipse(ColorWhite, 120, 240, 40, 30);
    fill_circle(ColorBlue, 67, 260, 10);
    fill_circle(ColorBlue, 132, 260, 10);       
    fill_triangle(ColorRed, 100, 240, 85, 280, 115, 280);   
    draw_triangle(ColorBlack, 100, 240, 85, 280, 115, 280); 
    fill_triangle(ColorRed, 100, 290, 100, 310, 70, 300);   
    fill_triangle(ColorRed, 100, 290, 100, 310, 130, 300);  
    draw_triangle(ColorBlack, 100, 290, 100, 310, 70, 300); 
    draw_triangle(ColorBlack, 100, 290, 100, 310, 130, 300);    

    refresh_screen();

    delay(1000);
    
    //Fourth House
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();
    delay(200);
    
    //Draw the sun
    fill_circle(ColorYellow, 50.0, 50.0, 20);
    draw_circle(ColorRed, 50.0, 50.0, 15);
    draw_circle(ColorRed, 50.0, 50.0, 12);
    draw_circle(ColorRed, 50.0, 50.0, 9);
    draw_circle(ColorRed, 50.0, 50.0, 6);
    fill_circle(ColorWhite, 50.0, 50.0, 3);
    draw_line(ColorYellow, 50, 75, 50, 110);
    draw_line(ColorYellow, 25, 75, 0, 100);
    draw_line(ColorYellow, 75, 75, 100, 100);
    draw_line(ColorYellow, 0, 50, 25, 50);
    draw_line(ColorYellow, 75, 50, 100, 50);
    draw_line(ColorYellow, 25, 25, 0, 0);
    draw_line(ColorYellow, 75, 25, 100, 0);
    draw_line(ColorYellow, 50, 0, 50, 25);
    refresh_screen();
    delay(200);

    //Draw Face
    fill_circle(ColorPink, 100, 250, 80);
    fill_ellipse(ColorWhite, 40, 240, 40, 30);
    fill_ellipse(ColorWhite, 120, 240, 40, 30);
    fill_circle(ColorBlue, 67, 260, 10);
    fill_circle(ColorBlue, 132, 260, 10);       
    fill_triangle(ColorRed, 100, 240, 85, 280, 115, 280);   
    draw_triangle(ColorBlack, 100, 240, 85, 280, 115, 280); 
    fill_triangle(ColorRed, 100, 290, 100, 310, 70, 300);   
    fill_triangle(ColorRed, 100, 290, 100, 310, 130, 300);  
    draw_triangle(ColorBlack, 100, 290, 100, 310, 70, 300); 
    draw_triangle(ColorBlack, 100, 290, 100, 310, 130, 300);    
    refresh_screen();
    delay(200);

    //Draw House
    fill_triangle(ColorRed, 300, 160, 200, 260, 400, 260);
    fill_rectangle(ColorGrey, 210, 260, 180, 180);
    fill_rectangle(ColorBlack, 215, 300, 80, 140);
    fill_rectangle(ColorLightGrey, 300, 300, 80, 60);
    fill_triangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    fill_triangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    fill_circle(ColorYellow, 280, 370, 3);
    refresh_screen();

    delay(1000);
    
    //Fifth Tree
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();
    delay(200);
    
    //Draw the sun
    fill_circle(ColorYellow, 50.0, 50.0, 20);
    draw_circle(ColorRed, 50.0, 50.0, 15);
    draw_circle(ColorRed, 50.0, 50.0, 12);
    draw_circle(ColorRed, 50.0, 50.0, 9);
    draw_circle(ColorRed, 50.0, 50.0, 6);
    fill_circle(ColorWhite, 50.0, 50.0, 3);
    draw_line(ColorYellow, 50, 75, 50, 110);
    draw_line(ColorYellow, 25, 75, 0, 100);
    draw_line(ColorYellow, 75, 75, 100, 100);
    draw_line(ColorYellow, 0, 50, 25, 50);
    draw_line(ColorYellow, 75, 50, 100, 50);
    draw_line(ColorYellow, 25, 25, 0, 0);
    draw_line(ColorYellow, 75, 25, 100, 0);
    draw_line(ColorYellow, 50, 0, 50, 25);
    refresh_screen();
    delay(200);

    //Draw Face
    fill_circle(ColorPink, 100, 250, 80);
    fill_ellipse(ColorWhite, 40, 240, 40, 30);
    fill_ellipse(ColorWhite, 120, 240, 40, 30);
    fill_circle(ColorBlue, 67, 260, 10);
    fill_circle(ColorBlue, 132, 260, 10);       
    fill_triangle(ColorRed, 100, 240, 85, 280, 115, 280);   
    draw_triangle(ColorBlack, 100, 240, 85, 280, 115, 280); 
    fill_triangle(ColorRed, 100, 290, 100, 310, 70, 300);   
    fill_triangle(ColorRed, 100, 290, 100, 310, 130, 300);  
    draw_triangle(ColorBlack, 100, 290, 100, 310, 70, 300); 
    draw_triangle(ColorBlack, 100, 290, 100, 310, 130, 300);    
    refresh_screen();
    delay(200);

    //Draw House
    fill_triangle(ColorRed, 300, 160, 200, 260, 400, 260);
    fill_rectangle(ColorGrey, 210, 260, 180, 180);
    fill_rectangle(ColorBlack, 215, 300, 80, 140);
    fill_rectangle(ColorLightGrey, 300, 300, 80, 60);
    fill_triangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    fill_triangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    fill_circle(ColorYellow, 280, 370, 3);
    refresh_screen();
    delay(200);

    //Draw Tree
    fill_triangle(ColorGreen, 500, 75, 440, 150, 560, 150);         
    fill_ellipse(ColorTurquoise, 400, 380, 200, 100);
    fill_rectangle(ColorBlack, 495, 274, 20, 160);
    fill_triangle(ColorGreen, 500, 120, 400, 275, 600, 275);
    fill_circle(ColorRed, 490, 150.0, 3);
    fill_circle(ColorRed, 510, 170, 4);
    fill_circle(ColorRed, 480, 240, 3);
    fill_circle(ColorRed, 560, 260, 5);
    refresh_screen();

    delay(1000);
    
    //Sixth Flats
    //Draw the background
    fill_rectangle(ColorBlue, 0, 0, 800, 300);
    fill_rectangle(ColorGreen, 0, 300, 800, 300);
    refresh_screen();
    delay(200);
    
    //Draw the sun
    fill_circle(ColorYellow, 50.0, 50.0, 20);
    draw_circle(ColorRed, 50.0, 50.0, 15);
    draw_circle(ColorRed, 50.0, 50.0, 12);
    draw_circle(ColorRed, 50.0, 50.0, 9);
    draw_circle(ColorRed, 50.0, 50.0, 6);
    fill_circle(ColorWhite, 50.0, 50.0, 3);
    draw_line(ColorYellow, 50, 75, 50, 110);
    draw_line(ColorYellow, 25, 75, 0, 100);
    draw_line(ColorYellow, 75, 75, 100, 100);
    draw_line(ColorYellow, 0, 50, 25, 50);
    draw_line(ColorYellow, 75, 50, 100, 50);
    draw_line(ColorYellow, 25, 25, 0, 0);
    draw_line(ColorYellow, 75, 25, 100, 0);
    draw_line(ColorYellow, 50, 0, 50, 25);
    refresh_screen();
    delay(200);

    //Draw Face
    fill_circle(ColorPink, 100, 250, 80);
    fill_ellipse(ColorWhite, 40, 240, 40, 30);
    fill_ellipse(ColorWhite, 120, 240, 40, 30);
    fill_circle(ColorBlue, 67, 260, 10);
    fill_circle(ColorBlue, 132, 260, 10);       
    fill_triangle(ColorRed, 100, 240, 85, 280, 115, 280);   
    draw_triangle(ColorBlack, 100, 240, 85, 280, 115, 280); 
    fill_triangle(ColorRed, 100, 290, 100, 310, 70, 300);   
    fill_triangle(ColorRed, 100, 290, 100, 310, 130, 300);  
    draw_triangle(ColorBlack, 100, 290, 100, 310, 70, 300); 
    draw_triangle(ColorBlack, 100, 290, 100, 310, 130, 300);    
    refresh_screen();
    delay(200);

    //Draw House
    fill_triangle(ColorRed, 300, 160, 200, 260, 400, 260);
    fill_rectangle(ColorGrey, 210, 260, 180, 180);
    fill_rectangle(ColorBlack, 215, 300, 80, 140);
    fill_rectangle(ColorLightGrey, 300, 300, 80, 60);
    fill_triangle(ColorYellow, 345, 300, 380, 300, 380, 355);
    fill_triangle(ColorYellow, 300, 300, 300, 355, 335, 300);
    fill_circle(ColorYellow, 280, 370, 3);
    refresh_screen();
    delay(200);

    //Draw Tree
    fill_triangle(ColorGreen, 500, 75, 440, 150, 560, 150);         
    fill_ellipse(ColorTurquoise, 400, 380, 200, 100);
    fill_rectangle(ColorBlack, 495, 274, 20, 160);
    fill_triangle(ColorGreen, 500, 120, 400, 275, 600, 275);
    fill_circle(ColorRed, 490, 150.0, 3);
    fill_circle(ColorRed, 510, 170, 4);
    fill_circle(ColorRed, 480, 240, 3);
    fill_circle(ColorRed, 560, 260, 5);
    refresh_screen();
    delay(200);

    //Draw the block
    fill_rectangle(ColorRed, 620, 50, 160, 450);
    fill_rectangle(ColorBlack, 640, 75, 40, 60);
    fill_rectangle(ColorLightGrey, 720, 75, 40, 60);
    fill_rectangle(ColorGrey, 640, 225, 40, 60);
    fill_rectangle(ColorYellow, 720, 225, 40, 60);
    fill_rectangle(ColorBlack, 670, 380, 60, 120);
    refresh_screen();
    
    delay(2000);

    release_all_resources();
    return 0;
}
